enhancements/microshift/advertise-version-details.md

```diff
@@ -0,0 +1,272 @@
+
+## Summary
+
+We have several use cases for allowing authenticated users to discover
+whether they are communicating with OCP or MicroShift, including the
+version of the product. We already have the ClusterVersion resource
+for OCP, but that requires special privileges to read and is not
+present in a MicroShift deployment. This enhancement describes a
+standard that both platforms can use to share platform and version
+information with API callers.
+
+## Motivation
+
+Users configuring applications to deploy on different versions of
+OpenShift may need to adjust their configuration to work differently
+on OCP and MicroShift. Ideally this is not needed, but in practice it
+is useful because of different APIs being available and different
+constraints being in place. It is possible for a cluster admin to read
+the ClusterVersion resource in an OCP cluster to discover that it is
+OCP and what version is running. This approach has two
+drawbacks. First, not all console users have cluster admin privileges,
+which means that the console cannot show them the cluster version
+details. Second, the ClusterVersion API is not present in a MicroShift
+deployment, so instead [we created a
+ConfigMap](https://github.com/openshift/microshift/pull/776) in the
+`openshift-public` namespace containing some basic version
+details. That means a caller must know two different APIs to check the
+version and product they are communicating with. This enhancement
+describes how we can standardize on one to make it easier for all
+authenticated users to obtain basic version details for a cluster.
+
+### User Stories
+
+As a console user, I want to see the version of OCP I am using so that
+I can refer to the correct version of the documentation and API
+references.
+
+As an authenticated API caller, I want to obtain the product name and
+version of the OpenShift variant I am communicating with so that I can
+use it to make decisions on the client side.
+
+### Goals
+
+* Avoid introducing a new CRD, so we do not increase the overhead for
+  MicroShift.
+* Use a standard Kubernetes type, to keep access easy for arbitrary
+  API clients.
+* Define a standard location for version information that all
+  OpenShift variants can support.
+
+### Non-Goals
+
+* This change does not introduce a new way to update the version of a
+  cluster.
+* This change does not introduce a new way for a client to discover
+  that a cluster is being upgraded.
+* This change does not introduce a new way for a client to discover
+  the cluster topology (single-node, HA, external control-plane, etc.).
+
+## Proposal
+
+Because we want arbitrary clients to be able to read and understand
+the data, we will use a standard Kubernetes ConfigMap.
+
+Because we want all OpenShift variants to publish the information in
+the same way and that the data should not be writable by API callers,
+we will specify that the ConfigMap should be named `version` and
+placed in the `openshift` namespace.
```

Comment:
`openshift-config-managed` might be more suitable

Reply:
We don't have that namespace in MicroShift right now.

```
$ oc get namespace
NAME                            STATUS   AGE
default                         Active   8d
kube-node-lease                 Active   8d
kube-public                     Active   8d
kube-system                     Active   8d
kubevirt-hostpath-provisioner   Active   8d
openshift                       Active   8d
openshift-controller-manager    Active   8d
openshift-dns                   Active   8d
openshift-infra                 Active   8d
openshift-ingress               Active   8d
openshift-node                  Active   8d
openshift-service-ca            Active   8d
```

Reply:
might be worth calling that out in the doc then, as to why `openshift` was chosen as the NS for this.

Reply:
Done in abba4340